spec/visual/samples/solidity

```diff
@@ -0,0 +1,242 @@
+
+            // TODO: highlight all known internal keywords?
+            tmp.origin = tx.origin;
+            tmp.result = garbled;
+            tmp.sender = msg.sender;
+            tmp.touched = true;
+
+            /* does this link-by-reference as expected?.. */
+            touchedMe[msg.sender] = tmp;
+            touchedMe[tx.origin] = tmp;
+        }
+        else {
+            /* weak guard against re-entry */
+            touchedMe[k].touched = false;
+            
+            opportunisticSend(k);
+            
+            delete touchedMe[k];
+            /* these probably do nothing... */
+            delete touchedMe[msg.sender];
+            delete touchedMe[tx.origin];
+        }
+    }
+}
+
+/**
+   \brief Examples of bad practices.
+
+   TODO: This special doxygen natspec notation is not parsed yet.
+
+   @author Noel Maersk
+ */
+/// TODO: Neither is this one.
+
+contract BadPractices {
+    address constant creator; /* `internal` by default */
+    address private owner; /* forbid inheritance */
+    bool mutex;
+
+    modifier critical {
+        assert(!mutex);
+        mutex = true;
+        _;
+        mutex = false;
+    }
+    
+    constructor() external {
+        creator = tx.origin;
+        owner = msg.sender;
+    }
+
+    /* Dangerous - function public, and doesn't check who's calling. */
+    function withdraw(uint _amount)
+        public
+        critical
+        returns (bool)
+    { /* `mutex` set via modifier */
+        /* Throwing on failed call may be dangerous. Consider
+           returning false instead?.. */
+        require(msg.sender.call.value(_amount)());
+        return true;
+    } /* `mutex` reset via modifier */
+    
+    /* fallback */
+    fallback() external payable {
+        /* `i` will be `uint8`, so this is an endless loop
+           that will consume all gas and eventually throw.
+         */
+        for (var i = 0; i < 257; i++) {
+            owner++;
+        }
+    }
+
+    /* receive()?.. nah, why bother */
+}
+
+/* /* /* /* /* /* /* /* /* /* /* /* /* /* /* /* /* /* /* /* /*
+// Open comment to EOF. Compiler chokes on this, but it's useful for
+// highlighting to show that there's an unmatched multi-line comment
+// open.
+//
+// On the other hand, a regular multi-line comment closure, including an
+// escaped variant as demonstrated shortly, should close the comment;
+// note that there are no nested multi-line comments.
+//
+// If the comment is still shown as "open", then a
+//
+//                !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+//                !!! MALICIOUS CODE SEGMENT !!!
+//                !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+//
+// can be erroneously thought of as inactive, and left unread.
+// In fact, the compiler will produce executable code it, possibly
+// overriding the program above.
+//
+// It is imperative that syntax highlighters do parse it if either of
+// `* /` or `\* /` (with space removed) are present.
+//
+// Now, let's party! :) \* /
```

Comment:
Just one final question: this is meant to be left _unclosed_, right? The text in the comment made it sound like maybe this was supposed to be closed.

Reply:
I did intentionally leave it unclosed now: that shows that open-comment-to-EOF is highlighted as described, and "guarantees" the code can't be compiled.

But it could be closed if you think that's how the text primarily reads. That, too, would be correct.

Should have probably split it in two... "Oh well."